### PR TITLE
Add version info to Windows executable.

### LINKED ISSUE #359: Missing version number in executable
We use revision detection for install/upgrading and it looks like the latest release made it out without any file version information. (Installer and uninstaller both show proper versions)

TL;DR: graylog-sidecar.exe is missing File version, Product Name, Product version, Copyright and language.

### Steps to reproduce the problem
1. ... Upgrade to 1.0.1.1

### PR DESCRIPTION
**Pull request by malcyon**
Adds version info to `graylog-sidecar` Windows executable using [goversioninfo](https://github.com/josephspurrier/goversioninfo)

Fixes #359

### CODE REVIEW

**Comment by malcyon** [comment]
Seems to work fine if I run `make build-windows32`.

image

But when I run `make build windows`, it throws this error:

```
donald@laptop:~/src/collector-sidecar$ make build-windows
go generate
GOOS=windows GOARCH=amd64 CGO_ENABLED=1 CC=x86_64-w64-mingw32-gcc go build -ldflags "-s -X github.com/Graylog2/collector-sidecar/common.GitRevision=0e9a808 -X github.com/Graylog2/collector-sidecar/common.CollectorVersion=1.2.0 -X github.com/Graylog2/collector-sidecar/common.CollectorVersionSuffix=-SNAPSHOT" -pkgdir /home/donald/go/go_win -v -i -o build/1.2.0/windows/amd64/graylog-sidecar.exe
# github.com/Graylog2/collector-sidecar
/usr/lib/go-1.13/pkg/tool/linux_amd64/link: running x86_64-w64-mingw32-gcc failed: exit status 1
/usr/bin/x86_64-w64-mingw32-ld: i386 architecture of input file `/tmp/go-link-134204588/000000.o' is incompatible with i386:x86-64 output
collect2: error: ld returned 1 exit status

make: *** [Makefile:72: build-windows] Error 2
```

I don't really know golang, so it's possible I made some serious misstep somewhere.

**Comment by malcyon** [comment]
Tested successfully on a Windows 10 VM, using the EXE from the Jenkins build:

image

**Comment by malcyon** [comment]
Just need to do something with `versioninfo.json`, so that we don't have to update the version in multiple places with each release.

**Comment by malcyon** [comment]
Here is what it looks like using the command line flags instead of the json file:

image

File: Makefile [hunk 1]
```diff
@@ -66,14 +68,20 @@ build-freebsd:
...
 
-build-windows32: ## Build sidecar binary for Windows 32bit
+build-windows32: install-goversioninfo ## Build sidecar binary for Windows 32bit
 	@mkdir -p build/$(COLLECTOR_VERSION)/windows/386
+	$(GOVERSIONINFO_BIN) -product-version="$(COLLECTOR_VERSION)-$(COLLECTOR_REVISION)" -ver-major="$(COLLECTOR_VERSION_MAJOR)" -product-ver-minor="$(COLLECTOR_VERSION_MINOR)" -product-ver-patch="$(COLLECTOR_VERSION_PATCH)" -product-ver-build="$(COLLECTOR_REVISION)" -file-version="$(COLLECTOR_VERSION)-$(COLLECTOR_REVISION)" -ver-major="$(COLLECTOR_VERSION_MAJOR)" -ver-minor="$(COLLECTOR_VERSION_MINOR)" -ver-patch="$(COLLECTOR_VERSION_PATCH)" -ver-build="$(COLLECTOR_REVISION)"
 	GOOS=windows GOARCH=386 CGO_ENABLED=1 CC=i686-w64-mingw32-gcc $(GO) build $(BUILD_OPTS) -pkgdir $(GOPATH)/go_win32 -v -i -o build/$(COLLECTOR_VERSION)/windows/386/graylog-sidecar.exe
 
+## Adds version info to Windows executable
+install-goversioninfo:
+	go install github.com/josephspurrier/goversioninfo/cmd/goversioninfo@latest
```
Comment (mpfz0r) [inline Makefile:83]: This only works with go1.16 and above.
But I can live with that.

https://go.dev/doc/go1.16#go-command

**Review by mpfz0r — APPROVED** [review]
LGTM

**Pull request merged**